server/server.js

```diff
@@ -25,20 +23,67 @@ const getAvailableDates = function(propertyId, fromDate, toDate) {
       date: {
         [Op.between]: [fromDate, toDate],
       }
-      
+    }
+  });
+};
+
+const getRoomPrice = function(propertyId) {
+  return db.Property.findOne({
+    where: {
+      id: propertyId
     }
   });
 };
 
 app.get('/price/:propertyId', function(req, res) {
-  let dates = getAvailableDates(propertyId, req.body.fromDate, req.body.toDate);
+  let fromDate = req.query.fromDate;
+  let toDate = req.query.toDate;
+  let dates;
+  let propertyId = req.params.propertyId;
   
-  if (!!req.body.fromDate && !!req.body.toDate) {
+  if (!!fromDate && !!toDate) {
+    dates = getAvailableDates(propertyId, fromDate, toDate);
 
-  } else if (!!req.body.fromDate) {
+  } else if (!!fromDate) {
+    // missing toDate, fill it in for 30 days after fromDate
+    let toDate = new Date(fromDate);
+    toDate.setDate(toDate.getDate() + 30);
+    dates = getAvailableDates(propertyId, fromDate, toDate);
 
   } else {
-
+    // missing fromDate and toDate, fill them in from today to 34 days from now. (to fit a 35 day calendar month)
+    let toDate = new Date();
+    toDate.setDate(toDate.getDate() + 34);
+    dates = getAvailableDates(propertyId, new Date(), toDate);
   }
-});
 
+  let price = getRoomPrice(propertyId);
+
+  dates.then((cal) => {
+    return cal.map((x) => {
+      return {
+        date: x.dataValues.date,
+        dayPriceModifier: x.dataValues.dayPriceModifier
+      };
+    });
+  })
+    .then((cal) => {
+      price.then((x) => {
+        return {
+          basePrice: x.dataValues.basePrice,
+          cityTax: x.dataValues.cityTax,
+          cleaningFee: x.dataValues.cleaningFee,
+          serviceFee: x.dataValues.serviceFee,
+          longStayDiscount: x.dataValues.longStayDiscount,
+          minStay: x.dataValues.minStay,
+          calendar: cal
+        };
+      })
+        .then((data) => {
+          res.json(data);
+        });
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
I think you can simply provide next as the final catch handler here and then Express will catch the error: https://expressjs.com/en/guide/error-handling.html